build.gradle

```diff
@@ -166,54 +168,64 @@ tasks.register('integrationTest', Test) {
+    }
     api 'com.exactpro.th2:grpc-common:3.11.1'
 
     implementation 'com.google.protobuf:protobuf-java-util'
     implementation 'com.exactpro.th2:grpc-service-generator:3.2.2'
-    implementation ("com.exactpro.th2:cradle-cassandra:${cradleVersion}")
+    implementation ("com.exactpro.th2:cradle-cassandra:${cradleVersion}") {
+        exclude group: 'org.slf4j', module: 'slf4j-log4j12' // because of the vulnerability
+    }
 
     //FIXME: Add these dependencies as api to grpc-... artifacts
     implementation "io.grpc:grpc-protobuf"
     implementation "io.grpc:grpc-core"
     implementation "io.grpc:grpc-netty"
-    implementation "io.grpc:grpc-services:1.32.1" //FIXME: add to bom
+    implementation "io.grpc:grpc-services"
 
     implementation "com.rabbitmq:amqp-client"
 
     implementation "org.jetbrains:annotations"
 
     implementation "org.apache.commons:commons-lang3"
-    implementation "org.apache.commons:commons-collections4:4.4"
+    api("org.apache.commons:commons-collections4:4.4") {
+        because('multimap in MqRouterFilterConfiguration')
+    }
     implementation "org.apache.commons:commons-text:1.9"
     implementation "commons-cli:commons-cli:1.5.0"
 
     implementation "com.fasterxml.jackson.core:jackson-core"
-    implementation "com.fasterxml.jackson.core:jackson-databind"
-    implementation "com.fasterxml.jackson.core:jackson-annotations"
+    api("com.fasterxml.jackson.core:jackson-databind") {
+        because('provide ability to use object mapper in components')
+    }
+    api("com.fasterxml.jackson.core:jackson-annotations") {
+        because('providee ability to use jackson annotations in components')
+    }
     implementation "com.fasterxml.jackson.datatype:jackson-datatype-jsr310"
     implementation "com.fasterxml.jackson.dataformat:jackson-dataformat-yaml"
-    // TODO: need to update jackson kotlin module to get rid of kotlin-reflect 1.3.72
     implementation "com.fasterxml.jackson.module:jackson-module-kotlin"
 
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:${kotlin_version}"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
 
     implementation 'com.fasterxml.uuid:java-uuid-generator:4.0.1'
 
     implementation 'io.github.microutils:kotlin-logging:2.1.21'
 
     implementation 'org.apache.logging.log4j:log4j-slf4j-impl'
     implementation 'org.apache.logging.log4j:log4j-1.2-api'
-    implementation 'org.apache.logging.log4j:log4j-api'
+    implementation 'org.apache.logging.log4j:log4j-core'
 
     implementation 'io.prometheus:simpleclient'
     implementation 'io.prometheus:simpleclient_hotspot'
     implementation 'io.prometheus:simpleclient_httpserver'
     implementation 'io.prometheus:simpleclient_log4j2'
-    implementation 'io.prometheus:simpleclient_log4j'
 
-    implementation 'io.fabric8:kubernetes-client:4.13.0'
+    // fix vulnerability in transitive dependency
+    implementation 'com.squareup.okhttp3:okhttp:4.10.0'
```

Comment:
Where it goes from? Please, add the reason why this is here in `bacause` block
```suggestion
    implementation ('com.squareup.okhttp3:okhttp:4.10.0') {
      because('fix vulnerability in transitive dependency for <original dependency name>')
    }
```